src/main/java/database/models/EntityStorageCache.java

```diff
@@ -137,6 +140,19 @@ public void invalidateCache(String recordId) {
         SqlHelper.deleteFromTableWhere(handler.getConnection(), TABLE_NAME, wherePair.first, wherePair.second);
     }
 
+    /**
+     * Removes cache records associated with the provided IDs
+     */
+    public void invalidateCaches(Collection<Integer> recordIds) {
```

Comment:
Other methods short circuit if `recordIds.size() == 0`. Would that be useful here?